pkg/core/image.go

```diff
@@ -125,7 +127,18 @@ func sha256Flattener(input string) string {
 	}
 	hasher := sha256.New()
 	hasher.Write([]byte(input))
-	return base + "-" + hex.EncodeToString(hasher.Sum(nil))
+	base, extension := extractExtension(base)
+	return base + "-" + hex.EncodeToString(hasher.Sum(nil)) + extension
+}
+
+func extractExtension(path string) (string, string) {
```

Comment:
I believe there could be a problem if `path` was something like `gonna.trick/you`.
Have a look at `filepath.Ext()`

Reply:
Thanks. I did not know about filepath.Ext and initially thought we were only dealing with filenames.

Reply:
Maybe we are. But someone is bound to use that func with something else one day or another...